enhancements/machine-api/aws-efa-networking.md

```diff
@@ -0,0 +1,253 @@
+
+> An Elastic Fabric Adapter (EFA) is a network device that you can attach to your Amazon EC2 instance to accelerate
+  High Performance Computing (HPC) and machine learning applications. EFA enables you to achieve the application
+  performance of an on-premises HPC cluster, with the scalability, flexibility, and elasticity provided by the AWS
+  Cloud.
+>
+> EFA provides lower and more consistent latency and higher throughput than the TCP transport traditionally used in
+  cloud-based HPC systems. It enhances the performance of inter-instance communication that is critical for scaling HPC
+  and machine learning applications. It is optimized to work on the existing AWS network infrastructure and it can
+  scale depending on application requirements.
+
+### Risks and Mitigations
+
+#### This feature cannot be leveraged for control plane instances
+
+Due to limitations of the libraries that the installer uses to create control plane machines, there is currently no
+way in which we can enable the control plane machines to be created day 0 with the EFA network interface type.
+
+As this feature is only intended to be used with specific high performance workloads, which should not be running on
+control plane hosts, we do not expect end users to require EFA network interfaces for their control plane machines.
+
+#### Users will have to configure their own security groups
+
+To allow the feature to work, the security groups that contain the Machines must be configured to allow all traffic
+between the hosts within the security group. This is not configured by default on an IPI cluster.
+Users must manually make this change via the AWS console/cli.
+This prerequisite should be documented as a manual step to enable the feature.
+
+Additionally, as we have future plans to manage security groups for IPI clusters, these manual changes to the security
+groups may be difficult to adopt into the future management system. However, there is nothing to stop users of IPI
+clusters making changes such as this already, so this kind of change must be considered for any adoption plans already.
+
+Additionally, IPI users are free to add additional security groups to their clusters post install and specify these
+additional groups within their Machine specs. We have no control over this today and again, this is something that must
+be planned for in the future security group management process.
+
+#### Using EFA requires additional software
+
+The EFA network interface types require special drivers to enable them to be used.
+Fortunately, the RHEL kernel already includes the relevant kernel modules to allow the adapter to be used transparently
+as a standard network adapter within user space. This means that no additional configuration is required within RHCOS
+to allow EFA instances to be used as regular OpenShift nodes, existing networking is unaffected.
+
+To take advantage of the EFA performance benefits, additional libfabric software must be deployed.
```

Comment:
Will enabling this cause a network hiccup on the workers?  Can we detect that this is enabled?  How will an author creating pods that use this feature know that the node has fully enabled the EFA feature?  This is also important for your test plan below.

Reply:
The documentation suggests that using MPI/libfabric has no effect at all on non-MPI/libfabric networking (except to the extent that it implies you're going to be using lots of inter-node bandwidth for MPI traffic, which will have the same effect on node networking as any other workload using lots of of inter-node bandwidth would).

Reply:
I think we can add an assumption (and document) that this configuration is expected to be done before there are active workloads running on those nodes.